kafka-connect/src/main/java/io/tabular/iceberg/connect/IcebergSinkTask.java

```diff
@@ -75,114 +50,57 @@ public void start(Map<String, String> props) {
     this.config = new IcebergSinkConfig(props);
   }
 
-  @Override
-  public void open(Collection<TopicPartition> partitions) {
-    // destroy any state if KC re-uses object
-    clearObjectState();
-    catalog = Utilities.loadCatalog(config);
```

Comment:
I don't see why you need two. 

re: passing SinkTaskContext and IcebergSinkConfig on the interface is fine.  I see no problem with passing a Catalog to the `*Impl` classes and avoiding it on the interface.  I don't see what is "wrong" about that. 

Regarding catalogs being thread safe or not... I don't think there is much of an issue --if there was a problem, we would have encountered it in the old code. 

imho 1 catalog is fine, don't see the need to create two.

Reply:
> I see no problem with passing a Catalog to the `*Impl` classes and avoiding it on the interface.

That's my problem essentially. When we make this `Committer` interface pluggable, I don't want to rely on  implementations having a constructor method that accepts a `Catalog` argument. In my head, it makes sense that `Committer` implementations will accept only two arguments; SinkTaskContext and IcebergSinkConfig. 

> Regarding catalogs being thread safe or not... I don't think there is much of an issue --if there was a problem, we would have encountered it in the old code.

I agree that so far we haven't seen or heard of any such issues. 
But that's not really a good reason to ignore this, we might just be getting lucky. 
When we move to pluggable Committer interfaces, I worry we might not be so lucky. 
Even in today's code, it is definitely possible for two threads to use the `Catalog` at the same time on leader tasks (`CoordinatorThread` and the regular thread).

Reply:
In the interests of scoping this PR down to just the essential changes required for enabling a pluggable committer interface, I have reverted this change so each SinkTask instance will still initialize only 1 Catalog (regardless of whether it is a leader task or not). 

I still have (maybe-overly-paranoid) thread-safety concerns about this but since we haven't seen any issues, it seems fine to leave things as-is.